### PR TITLE
minor cleanup

### PR DESCRIPTION
**Pull request by element-of-surprise**
Removes some minor functions out of msalbase that can easily be done directly with a few more characters.  This removes more dependencies on the msalbase package as we start thinking on integration with msal/

Also removes some unused functions.

### CODE REVIEW

File: internal/storage/storage.go [hunk 1]
```diff
@@ -51,7 +52,7 @@ func checkAlias(alias string, aliases []string) bool {
 }
 
 func isMatchingScopes(scopesOne []string, scopesTwo string) bool {
-	newScopesTwo := msalbase.SplitScopes(scopesTwo)
+	newScopesTwo := strings.Split(scopesTwo, " ")
```
Comment (jhendrixMSFT) [inline internal/storage/storage.go:55]: Was this supposed to be `scopeSeparator` here?

File: internal/msalbase/internal.go [hunk 2]
```diff
@@ -18,7 +18,6 @@ import (
 	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/json"
 	jwt "github.com/dgrijalva/jwt-go"
 	uuid "github.com/google/uuid"
-	"github.com/shirou/gopsutil/host"
```
Comment (keegan-caruso) [inline internal/msalbase/internal.go:21]: This looks like the only usage of this package, can we take it out of the go.mod/go.sum?

File: internal/msalbase/internal.go [hunk 3]
```diff
@@ -178,20 +177,6 @@ func CreateOAuthResponseBase(httpStatusCode int, responseData []byte) (OAuthResp
...
-	return h.Platform
-}
-
-// GetOSVersion gets the OS version that the client using MSAL is running on.
-// TODO(jdoak): Remove this.
-func GetOSVersion() string {
-	h, _ := host.Info()
-	return h.PlatformVersion
-}
-
```
[the remark this answers is not in the document]
Comment (element-of-surprise) [reply internal/msalbase/internal.go:194]: They aren't used at all.  If someone adds telemetry, then we can honestly just do:

h, err := host.Info()
if err != nil {
  return nil, err
}
h.Platform
h.PlatformVersion

[the remark this answers is not in the document]
Comment (sangonzal) [reply internal/msalbase/internal.go:194]: I see. It's [here](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/2c86158d91b8e1f685c6a2affac07366d208de2e/msal/default_web_request_manager.go#L242), it's just never set. 

We need the top 4 headers listed [here](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/df65b5c0bb36c3fc674b4e350e6cb967f7b69763/src/main/java/com/microsoft/aad/msal4j/HttpHeaders.java#L13). If you don't think they belong in this PR I can open up an issue to track this work.

[the remark this answers is not in the document]
Comment (keegan-caruso) [reply internal/msalbase/internal.go:194]: x-client-OS and x-client-SKU are [set](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/2c86158d91b8e1f685c6a2affac07366d208de2e/msal/default_web_request_manager.go#L249)

x-client-cpu can be set by GOARCH

x-client-version will need to be set differently

[the remark this answers is not in the document]
Comment (element-of-surprise) [reply internal/msalbase/internal.go:194]: Let's put in a feature for this.

File: internal/msalbase/internal.go (hunk 2 above)
Comment (element-of-surprise) [reply internal/msalbase/internal.go:21]: We can, but we should do that with a go mod tidy in the future.

What do you think about you adding a check that the go.mod file is updated via tidy?

**Pull request merged**